src/baseConfig.js

```diff
@@ -0,0 +1,127 @@
+import extend from 'extend';
+
+/**
+ * BaseConfig class
+ */
+export class BaseConfig {
+
+  /**
+   * The defaults of your config
+   * @param {Object}
+   */
+  defaults = {};
+
+  /**
+   * The current values of your config
+   * @type {Object}
+   */
+  current = {};
+
+  /**
+   * Derived classes must set a unique alias
+   * @type {String}
+   */
+  alias
+
+  /**
+   * Creates an instance of BaseConfig. Copies defaults into current
+   * @param  {Object} defaultsOverwrite = {} Optionally extends the defaults
+   */
+  constructor(defaultsOverwrite = {}) {
```

Comment:
Overwriting the defaults should be a method `configureDefaults`. Or even better would be to make defaults a new BaseConfig instance. That way you leverage the BaseConfig class you already wrote.

```js
export class BaseConfig {

  constructor(defaults) {
    defaults && this.defaults = new BaseConfig();
    /* ... */
  }
```